executor/aggfuncs/func_percentile.go

```diff
@@ -39,7 +39,7 @@ var (
 
 func percentile(data sort.Interface, percent int) int {
 	// Ordinal rank k = Ceil(P / 100 * N)
-	k := int(math.Ceil(float64(data.Len()) / 100 * float64(percent)))
```

Comment:
Do the multiply first may cause overflow.

By the way, I've just filed a PR to fix this but find you've get it done 🤣 . https://github.com/pingcap/tidb/compare/master...leiysky:fix/approx_percentile?expand=1

My main idea is using decimal to calculate the precise `k` value here, FYI.

Reply:
good catch~ how about `p/100` first, then len * [a num <= 1] should not great then len 😆

Reply:
Sounds good to me.